trio/_ssl.py

```diff
@@ -685,22 +686,19 @@ def _check_status(self):
                 max_bytes = _operator.index(max_bytes)
                 if max_bytes < 1:
                     raise ValueError("max_bytes must be >= 1")
-            try:
-                return await self._retry(self._ssl_object.read, max_bytes)
-            except trio.BrokenResourceError as exc:
-                # This isn't quite equivalent to just returning b"" in the
-                # first place, because we still end up with self._state set to
-                # BROKEN. But that's actually fine, because after getting an
-                # EOF on TLS then the only thing you can do is close the
-                # stream, and closing doesn't care about the state.
-                if (
-                    self._https_compatible
-                    and isinstance(exc.__cause__, _stdlib_ssl.SSLEOFError)
-                ):
-                    await trio.hazmat.checkpoint()
+
+            def do_read():
```

Comment:
one thought here is that maybe it would be better to move this function out into it's own method? i don't have any really strong opinion on whether or not that makes that code more readable. but i just wanted to bring it up to see what others think. i definitely think that this is cleaner than the previous approach.